extensions/wikia/Wall/WallBaseController.class.php

```diff
@@ -557,18 +557,19 @@ public function brickHeader() {
 	}
 
 	public function newMessage() {
-		$wall_username = $this->helper->getUser()->getName();
+		$user = $this->helper->getUser();
+		$wallUsername = $user->getName();
 
 		// only use realname if user made edits (use logic from masthead)
-		$userStatsService = new UserStatsService( $this->helper->getUser()->getID() );
+		$userStatsService = new UserStatsService( $user->getID() );
 		$userStats = $userStatsService->getStats();
 		if ( empty( $userStats[ 'editcount' ] ) || $userStats[ 'editcount' ] == 0 ) {
-			$wall_username = $this->helper->getUser()->getName();
+			$wallUsername = $user->getName();
 		}
 
 		$username = $this->wg->User->getName();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app